src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +158,123 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
+
+        OPENVINO_ASSERT(batch.is_dynamic() || batch.get_length() == input_size,
+                        "set_input_tensors/set_tensors error. Input shape ",
+                        param->get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        input_size);
+    }
+    OPENVINO_ASSERT(batch_idx == 0,
+                    "set_input_tensors/set_tensors is not currently supported for batch dimension index ",
+                    batch_idx,
+                    " != 0");
+    // Initial implementation always performs creation of new Blob and 'memcpy'
+    // In future consider checking if blobs point to contiguous range of memory
+    auto& desc = blobs[0]->getTensorDesc();
+    desc.getDims()[batch_idx] = input_size;
+    auto blockingDims = desc.getBlockingDesc().getBlockDims();
+    blockingDims[batch_idx] = input_size;
+    auto blockingDesc = BlockingDesc(blockingDims,
+                                     desc.getBlockingDesc().getOrder(),
+                                     desc.getBlockingDesc().getOffsetPadding(),
+                                     desc.getBlockingDesc().getOffsetPaddingToData(),
+                                     desc.getBlockingDesc().getStrides());
+    auto batched_desc = InferenceEngine::TensorDesc(desc.getPrecision(), desc.getDims(), blockingDesc);
+    auto desc_to_string = [](const TensorDesc& desc) {
+        std::stringstream s;
+        s << "{ " << desc.getLayout() << " " << desc.getPrecision().name();
+        s << "dim=(";
+        for (const auto& d : desc.getDims()) {
+            s << " " << d;
+        }
+        s << " ) }";
+        return s.str();
+    };
+    std::for_each(blobs.begin(), blobs.end(), [&desc, &batch_idx, &desc_to_string](const Blob::Ptr& item) {
+        auto item_desc = item->getTensorDesc();
+        item_desc.getDims()[batch_idx] = desc.getDims()[batch_idx];
+        OPENVINO_ASSERT(item_desc.getDims() == desc.getDims() && item_desc.getLayout() == desc.getLayout() &&
+                            item_desc.getPrecision() == desc.getPrecision(),
+                        "set_input_tensors/set_tensors error. Blob ",
+                        desc_to_string(item_desc),
+                        " is not compatible with batched blob ",
+                        desc_to_string(desc));
+    });
+    // Create batched blob. TODO: no API to get allocator of original blobs, use default one
+    auto batched_blob = make_blob_with_precision(batched_desc);
+    batched_blob->allocate();
+    MemoryBlob::Ptr batched_mem_blob = as<MemoryBlob>(batched_blob);
+    auto ptr = batched_mem_blob->wmap();
+    auto u8_ptr = ptr.as<uint8_t*>();
+
+    // Perform memory copy
+    for (size_t i = 0; i < blobs.size(); i++) {
+        const auto& blob = as<MemoryBlob>(blobs[i]);
+        memcpy(u8_ptr + i * blob->byteSize(), blob->rmap().as<void*>(), blob->byteSize());
```

Comment:
one of the problem with it:
```c++
vector<Tensor> tensors(...);
r.set_input_tensors(tensors);
for (int i = 0; i < 10; ++i) {
  // update values in tensors
  r.infer(); // does not work correctly since does not take into account updates in tensors (they are shared_ptr-like objects)
}
```
Work with a single tensor in the above manner is allowed. Need to think should we move this "execution" on inference part..